analytics/upress_reporting/helpers/download_request_parser.py

```diff
@@ -0,0 +1,149 @@
+import os
+import boto3
+import numpy
+import re
+import shutil
+
+from logger import createLog
+from model import Edition, Item, Link
+from model.postgres.item import ITEM_LINKS
+from managers import DBManager
+
+# Regexes needed to parse S3 logs
+REQUEST_REGEX = r"REST.GET.OBJECT "
+# File ID includes the file name for the pdf object
+FILE_ID_REGEX = r"REST.GET.OBJECT (.+pdf\s)"
+TIMESTAMP_REGEX = r"\[.+\]"
+REFERRER_REGEX = r"https://drb-qa.nypl.org/"
+
+
+class DownloadRequestParser:
+    """
+    Parses S3 download logs and generates CSVs with all download requests
+    for each day. All CSVs are stored locally in "download_csvs" directory.
+    """
+
+    def __init__(self, publisher, date_range):
+        self.publisher = publisher
+        self.date_range = date_range
+
+        self.s3_client = boto3.client("s3")
+        self.bucket_name = os.environ.get("DOWNLOAD_BUCKET", None)
+        self.log_path = os.environ.get("DOWNLOAD_LOG_PATH", None)
+
+        self._setup_db_manager()
+        self.logger = createLog("download_request_parser")
+
+    def generate_csv_files(self):
```

Comment:
To recap our discussion, my recommendation is to change this function to return a list of downloads which is a combination of a counter 5 representation of a book/work/title + when the book was downloaded. 

Let's start small and just return the title and when the download occurred as part of this report.